api/agent/agent.go

```diff
@@ -119,35 +121,30 @@ type agent struct {
 
 // New creates an Agent that executes functions locally as Docker containers.
 func New(da DataAccess) Agent {
-	a := createAgent(da, true, nil).(*agent)
+	a := createAgent(da, nil).(*agent)
 	if !a.shutWg.AddSession(1) {
 		logrus.Fatalf("cannot start agent, unable to add session")
 	}
 	go a.asyncDequeue() // safe shutdown can nanny this fine
 	return a
 }
 
-func createAgent(da DataAccess, withDocker bool, withShutWg *common.WaitGroup) Agent {
+func createAgent(da DataAccess, withShutWg *common.WaitGroup) Agent {
```

Comment:
there are no callers that call this with a non-nil `withShutWg` now, can we nuke that one too while we're here?